engine/include/cubos/engine/gizmos/gizmos.hpp

```diff
@@ -0,0 +1,89 @@
+/// @file
+/// @brief Resource @ref cubos::engine::Gizmos.
+/// @ingroup gizmos-plugin
+
+#pragma once
+
+#include <vector>
+
+#include <glm/vec3.hpp>
+
+#include <cubos/core/gl/render_device.hpp>
+
+#include <cubos/engine/cubos.hpp>
+
+namespace cubos::engine
+{
+    class GizmosRenderer;
+
+    /// @brief Resource which queues for drawing gizmos, basic primitives useful for debugging and tools.
+    ///
+    /// @ingroup gizmos-plugin
+    class Gizmos final
+    {
+    public:
+        /// @brief Sets the color to be used when drawing any subsequent gizmos.
+        /// @param color Color to be used.
+        void color(const glm::vec3& color);
+
+        /// @brief Draws a line gizmo.
+        /// @param id Identifier of the gizmo.
+        /// @param from One of the ends of the line to be drawn.
+        /// @param to The other end of the line to be drawn.
+        /// @param lifespan How long the line will be on screen for, in seconds. Defaults to 0, which means a single
+        /// frame.
+        void drawLine(const std::string& id, glm::vec3 from, glm::vec3 to, float lifespan = 0.0F);
+
+        /// @brief Draws a filled box gizmo.
+        /// @param id Identifier of the gizmo.
+        /// @param corner One of the corners of the box to be drawn.
+        /// @param oppositeCorner The opposite corner of the box to be drawn.
```

Comment:
Why not from/to for consistency?

Reply:
a box doesn't really go from a point to another like a line does